hack/generator/pkg/astmodel/package_import_set.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import "sort"
+
+// PackageImportSet represents a set of distinct PackageImport references
+type PackageImportSet struct {
```

Comment:
Would it be better to just make this:
`type PackageImportSet map[PackageImport]struct{}`? We do this for other set types (for example `Types`, `TypeNameSet`)

Reply:
If you did this you could also drop the `*` everywhere on this type since `map`'s are always pointers.

Reply:
There was a reason why that didn't work - but I don't recall it. I'll spike the change and will use it if it works.

Reply:
Changed to use `map[PackageImport]struct{}`.